packages/core/saved-objects/core-saved-objects-migration-server-internal/src/next.ts

```diff
@@ -178,7 +190,7 @@ export const nextActionMap = (client: ElasticsearchClient, transformRawDocs: Tra
       Actions.bulkOverwriteTransformedDocuments({
         client,
         index: state.targetIndex,
-        transformedDocs: state.transformedDocBatches[state.currentBatch],
+        operations: state.bulkOperationBatches[state.currentBatch],
```

Comment:
I was thinking, we no longer ONLY update documents in this step, but we delete them as well.
What happens if there are conflicts...

- ...document updated by another instance? We have optimistic concurrency control for that one.
- ...document deleted by another instance? does this cause the bulk operation to fail? I don't see a `conflicts: 'proceed'` flag here.

Reply:
This part of the flow affects corrupt and transform errors, and it is run systematically even by up-to-date deployments, so if a deployment fails to delete some of them it will retry on next startup.

The question here is whether it is safe to leave some corrupt and un-transformed documents in the system indices, and start "normally". WDYT @rudolf ?

Reply:
good you raised this... I think the biggest problem is if delete "conflicts" cause the whole migration to fail. Even if a restart should fix everything seeing a `FATAL` error is never nice.

Like Ahmad explains deleteByQuery throws version conflict exceptions for a double delete. But if we don't specify a seq_no _primary_term I'm not sure that bulk delete would do the same. It would probably return a not found error so ideally we should handle that and ignore that in the model.

Reply:
The docs seem to indicate that such errors are collected and the bulk operation continues (we don't even have the `conflicts: 'proceed'` flag here), but I'll double check that to confirm. 

After sleeping through it, I realised the other consequence of this change is that we now perform this cleanup of corrupt and transform errors systematically at startup (even on up-to-date deployments), whereas we used to do that on version upgrade only. This is less true for the transform errors, given that in a normal scenario all documents' versions should match the current one.

And, as a matter of fact, this is an improvement over the current implementation, where corrupt docs and transform errors are not cleaned up, and the migrator currently ignores the `discardCorruptObjects` flag at this point (it will simply fail to start).